ci/upload_audit_events_s3.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+
+from urllib.parse import urljoin
+from botocore.exceptions import ClientError
+
+import json
+import boto3
+import requests
+import os
+import functools
+from datetime import datetime, timedelta, timezone
+
+
+s3_client = boto3.client("s3")
+timestamp_key = "timestamp"
+
+
+class AuditEventsS3Uploader:
+    def __init__(self):
+        self.CF_API_URL = os.environ.get("CF_API_URL")
+        self.UAA_API_URL = os.environ.get("UAA_API_URL")
+        self.UAA_CLIENT_ID = os.environ.get("UAA_CLIENT_ID")
+        self.UAA_CLIENT_SECRET = os.environ.get("UAA_CLIENT_SECRET")
+        self.bucket_name = "{}".format(os.environ["BUCKET"])
+        self.token = self.get_client_credentials_token()
+
+    def get_client_credentials_token(self):
+        with requests.Session() as s:
```

Comment:
I *think* you could make your header management (specifically for the token) easier if you made one `Session` that was a property on `AuditEventsS3Uploader` object.

Reply:
I did try that but I couldn't seem to get it right

Reply:
ah, ok. The only advantage (I think) is the convenience, so it's fine to leave if it's more work.